addons/l10n_tr_nilvera_einvoice/models/account_move.py

```diff
@@ -0,0 +1,255 @@
+import uuid
+from markupsafe import Markup
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.addons.l10n_tr_nilvera.lib.nilvera_client import _get_nilvera_client
+
+
+class AccountMove(models.Model):
+    _name = 'account.move'
+    _inherit = ['account.move']
+
+    l10n_tr_nilvera_uuid = fields.Char(
+        string="Nilvera Document UUID",
+        copy=False,
+        readonly=True,
+        default=lambda self: str(uuid.uuid4()),
+        help="Universally unique identifier of the Invoice",
+    )
+    l10n_tr_nilvera_send_status = fields.Selection(
+        selection=[
+            ('error', "Error (check chatter)"),
+            ('not_sent', "Not sent"),
+            ('sent', "Sent and waiting response"),
+            ('succeed', "Successful"),
+            ('waiting', "Waiting"),
+            ('unknown', "Unknown"),
+        ],
+        string="Nilvera Status",
+        readonly=True,
+        copy=False,
+        default='not_sent',
+    )
+
+    @api.model
+    def _get_ubl_cii_builder_from_xml_tree(self, tree):
+        customization_id = tree.find('{*}CustomizationID')
+        if customization_id is not None and 'TR1.2' in customization_id.text:
+            return self.env['account.edi.xml.ubl.tr']
+        return super()._get_ubl_cii_builder_from_xml_tree(tree)
+
+    def button_draft(self):
+        # EXTENDS account
+        for move in self:
+            if move.l10n_tr_nilvera_uuid and move.l10n_tr_nilvera_send_status != 'not_sent':
+                raise UserError(_("You cannot reset to draft an entry that has been sent to Nilvera."))
+        super().button_draft()
+
+    def _l10n_tr_nilvera_submit_einvoice(self, xml_file, customer_alias):
+        self._l10n_tr_nilvera_submit_document(
+            xml_file=xml_file,
+            endpoint=f"/einvoice/Send/Xml?Alias={customer_alias}",
+        )
+
+    def _l10n_tr_nilvera_submit_earchive(self, xml_file):
+        self._l10n_tr_nilvera_submit_document(
+            xml_file=xml_file,
+            endpoint="/earchive/Send/Xml",
+        )
+
+    def _l10n_tr_nilvera_submit_document(self, xml_file, endpoint):
```

Comment:
Hey, I was going through your PR again before pinging the security team and I noticed this (which I must have forgotten or overlooked mb). As we discussed, we don't want to allow users to send arbitrary XML to the authority. It is best to compute/generate the XML and send it in the same method.

Reply:
It can be regenerated, but this means we do this step twice. Is it worth it?